frontend/src/metabase/collections/containers/CollectionContent.jsx

```diff
@@ -123,8 +123,23 @@ function CollectionContent({ collection, collectionId, isAdmin, isRoot }) {
   };
 
   return (
-    <Search.ListLoader query={pinnedQuery} wrapped>
-      {({ list: pinnedItems }) => {
+    <Search.ListLoader
+      query={pinnedQuery}
+      loadingAndErrorWrapper={false}
+      keepPreviousList
+      wrapped
+    >
+      {({
+        list: pinnedItemsList = [],
+        previousList: previousPinnedItemsList = [],
+        loading: loadingPinnedItems,
+      }) => {
+        // Prevents table UI flashing while performing a GET request
+        // (e.g. for sorting)
+        const pinnedItems =
+          pinnedItemsList.length > 0
+            ? pinnedItemsList
+            : previousPinnedItemsList;
         const hasPinnedItems = pinnedItems.length > 0;
```

Comment:
If I got it right I'm not sure it can work. `hasPinnedItems` is passed to the bulk actions bar and is used to determine if we should show an empty state. The trick is `pinnedItemsList` will be empty if there is an ongoing GET request (e.g. when changing sorting), so this can break bulk selections

Reply:
oh yeah, I misread the constant name there.